src/net/sourceforge/kolmafia/session/QuestManager.java

```diff
@@ -633,6 +639,42 @@ private static void handleFarmChange(final String location, final String respons
     }
   }
 
+  private static void handleFriarsChange(final String location, final String responseText) {
+    String[] elbowNCs = {
+      "Deep Imp Act",
+      "Imp Art, Some Wisdom",
+      "A Secret, But Not the Secret You're Looking For",
+      "Butter Knife? I'll Take the Knife"
+    };
+    String[] neckNCs = {
+      "How Do We Do It? Quaint and Curious Volume!",
+      "Strike One!",
+      "Olive My Love To You, Oh.",
+      "Dodecahedrariffic!"
+    };
+    String[] heartNCs = {
+      "Moon Over the Dark Heart", "Running the Lode", "I, Martin", "Imp Be Nimble, Imp Be Quick"
+    };
+
+    switch (location) {
+      case "The Dark Elbow of the Woods":
+        if (Arrays.stream(elbowNCs).anyMatch((nc) -> responseText.contains(nc))) {
+          Preferences.setInteger("lastFriarsElbowNC", AdventureSpentDatabase.getTurns(location));
```

Comment:
This is the encounter name right? I imagine we have a function somewhere to extract it from response text

Reply:
That was based on other code I saw in the area, but I'll look around